cmd/backup-manager/app/backup/backup.go

```diff
@@ -14,200 +14,128 @@
+	BackupName string
 }
 
-func (bo *BackupOpts) getDestBucketURI(remotePath string) string {
-	return fmt.Sprintf("%s://%s", bo.StorageType, remotePath)
+func (bo *Options) String() string {
+	return fmt.Sprintf("%s/%s", bo.Namespace, bo.BackupName)
 }
 
-func (bo *BackupOpts) getTikvGCLifeTime(db *sql.DB) (string, error) {
-	var tikvGCTime string
-	sql := fmt.Sprintf("select variable_value from %s where variable_name= ?", constants.TidbMetaTable)
-	row := db.QueryRow(sql, constants.TikvGCVariable)
-	err := row.Scan(&tikvGCTime)
+func (bo *Options) backupData(backup *v1alpha1.Backup) (string, error) {
+	args, path, err := constructOptions(backup)
 	if err != nil {
-		return tikvGCTime, fmt.Errorf("query cluster %s %s failed, sql: %s, err: %v", bo, constants.TikvGCVariable, sql, err)
+		return "", err
 	}
-	return tikvGCTime, nil
-}
-
-func (bo *BackupOpts) setTikvGCLifeTime(db *sql.DB, gcTime string) error {
-	sql := fmt.Sprintf("update %s set variable_value = ? where variable_name = ?", constants.TidbMetaTable)
-	_, err := db.Exec(sql, gcTime, constants.TikvGCVariable)
+	var btype string
+	if backup.Spec.Type == "" {
+		btype = string(v1alpha1.BackupTypeFull)
+	} else {
+		btype = string(backup.Spec.Type)
+	}
+	fullArgs := []string{
+		"backup",
+		btype,
+	}
+	fullArgs = append(fullArgs, args...)
+	output, err := exec.Command("br", fullArgs...).CombinedOutput()
 	if err != nil {
-		return fmt.Errorf("set cluster %s %s failed, sql: %s, err: %v", bo, constants.TikvGCVariable, sql, err)
+		return path, fmt.Errorf("cluster %s, execute br command %v failed, output: %s, err: %v", bo, args, string(output), err)
 	}
-	return nil
+	glog.Infof("Backup data for cluster %s successfully, output: %s", bo, string(output))
+	return path, nil
 }
 
-func (bo *BackupOpts) dumpTidbClusterData() (string, error) {
-	bfPath := bo.getBackupFullPath()
-	err := util.EnsureDirectoryExist(bfPath)
+// getCommitTs get backup position from `EndVersion` in BR backup meta
+func getCommitTs(backup *v1alpha1.Backup) (uint64, error) {
+	var commitTs uint64
+	s, err := util.NewRemoteStorage(backup)
 	if err != nil {
-		return "", err
-	}
-	args := []string{
-		fmt.Sprintf("--outputdir=%s", bfPath),
-		fmt.Sprintf("--host=%s", bo.Host),
-		fmt.Sprintf("--port=%d", bo.Port),
-		fmt.Sprintf("--user=%s", bo.User),
-		fmt.Sprintf("--password=%s", bo.Password),
-		"--long-query-guard=3600",
-		"--tidb-force-priority=LOW_PRIORITY",
-		"--verbose=3",
-		"--regex",
-		"^(?!(mysql|test|INFORMATION_SCHEMA|PERFORMANCE_SCHEMA))",
+		return commitTs, err
 	}
-
-	output, err := exec.Command("/mydumper", args...).CombinedOutput()
+	defer s.Close()
+	ctx := context.Background()
+	exist, err := s.Exists(ctx, MetaFile)
 	if err != nil {
-		return bfPath, fmt.Errorf("cluster %s, execute mydumper command %v failed, output: %s, err: %v", bo, args, string(output), err)
+		return commitTs, err
 	}
-	return bfPath, nil
-}
+	if !exist {
+		return commitTs, fmt.Errorf("%s not exist", MetaFile)
```

Comment:
Add `bo.String()` info to error log

Reply:
Here is to return error, the caller will log the error with backup CR info.